pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -353,15 +354,37 @@ public void reloadSegment(String segmentName, IndexLoadingConfig indexLoadingCon
     indexLoadingConfig.setSegmentTier(segmentTier);
     indexLoadingConfig.setTableDataDir(_tableDataDir);
     indexLoadingConfig.setInstanceTierConfigs(_tableDataManagerConfig.getInstanceTierConfigs());
+    indexLoadingConfig.setSegmentTier(zkMetadata.getTier());
```

Comment:
This change may be problematic. At L354, `setSegmentTier(segmentTier)` is called in order to create the SegmentDirectory object with current tier to kick off the following segment reloading logic.

At L386, `indexLoadingConfig.setSegmentTier(zkMetadata.getTier());` is called in order to load the segment with the target tier from the segment ZK metadata to finish segment loading. Different tiers may have different configs, for example `TierBasedSegmentDirectoryLoader` moves segments across data paths as set in tierConfigs, based on which tier is provided via this `setSegmentTier`.

Reply:
Thanks for catching this. Patching this with creating a new directory when we're sure existing one can be reused. lmk what you think.